### DISCUSSION TITLE
how can i get this Encoding settings

### DISCUSSION THREAD

**Discussion by mr-qansuh** [staxrip/staxrip#540] [Q&A]
how can i get this Encoding settings 

Encoding settings           : cabac=1 / ref=9 / deblock=1:1:1 / analyse=0x3:0x133 / me=umh / subme=10 / psy=1 / psy_rd=0.40:0.00 / mixed_ref=1 / me_range=24 / chroma_me=1 / trellis=2 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=1 / chroma_qp_offset=-2 / threads=12 / lookahead_threads=3 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=10 / b_pyramid=2 / b_adapt=2 / b_bias=0 / direct=3 / weightb=1 / open_gop=0 / weightp=2 / keyint=96 / keyint_min=9 / scenecut=40 / intra_refresh=0 / rc_lookahead=60 / rc=crf / mbtree=1 / crf=22.0 / qcomp=0.60 / qpmin=0 / qpmax=69 / qpstep=4 / ip_ratio=1.40 / aq=1:0.60

**Comment by ghost** [comment]
is this x264 or x265 ? it seems x264 for me... 

How to get them ? open the encoder's window, and type everyone of them in the search box of that window, you'll be directed to the exact matching item. Set it.

Most of them are defaults... 🤣  (maybe ALL)

My friend you need to focus and understand better: sha88el rasak shway ya 5ayye.... 🤣

**Reply by mr-qansuh** [reply]
but i don't know the preset

**Reply by stax76** [reply]
Presets are documented here:

https://x265.readthedocs.io/en/master/presets.html

**Reply by Dendraspis** [reply]
The link is for x265, but it must be x264 regarding the parameters.  🤔

**Reply by JJKylee** [reply]
There's [this online document](http://www.chaneru.com/Roku/HLS/X264_Settings.htm) for `x264` but it's 10 years old. 😓
So I mostly refer to the console help for `x264`.

**Comment by stax76** [comment]
You can transform it into a command line with search and replace or multi caret editing, then import the command line in the encoding dialog which has an import feature.

**Reply by ghost** [reply]
@stax76  They are all default setting. if you put NOTHING, you get those settings

**Comment by JJKylee** [comment]
You can check out the _**basic encoding processes**_ to see which options were adopted by the original encoder.

1. Which _RC mode_ was used: quality(`crf`), bitrate 1pass (with/without cbr), bitrate 2pass, or quantizer(`qp`).

It is obvious from the example above that the encoder adopted a quality mode with `crf=22.0`.

2. Which _Preset_ fits best: _medium_ is the default, so you need to check if another preset than medium fits better.
3. Which _Profile_ fits best: _Profile_ changes some option values after _Preset_.
4. Which _Tune_ fits best: _Tune_ also changes some option values after _Preset_.
5. See if there are any remaining `x264` option changes that can't be covered by Steps 1~4.

You need to refer to the _**console**_ `x264` help in all these steps.
_It's a really tedious and cumbersome job._


Having said that, this approach of replicating the original encoder's parameter settings is totally **_meaningless_** and **_useless_**.
Why? Because there are obvious limits to this method:

1. All encoding jobs depend _definitively_ on the **_source_**. So even if a setting happens to be good for one source, it doesn't mean that the same setting is also good for another. You _must_ take into consideration what kind of features/properties the source has every time you do an encoding job.

2. **_Filter jobs_** cannot be figured out by analyzing the encoding settings metadata. In lots of cases, it is filter jobs that determine the final output, but this info is not written in the metadata at all.

So in my experience, people with at least some modest knowledge about encoding do not ask this kind of question. It is too costly and gives almost no gain.

**Reply by ghost** [reply]
Look @mr-qansuh 
I have already answered you 2 days ago to help you (click link below):
https://github.com/staxrip/staxrip/issues/530#issuecomment-783456418

Maybe you didn't read it because it's after it was closed. So read it NOW.
